File: programmers/가장 큰 수.cpp
#include <vector>
#include <string>
#include <algorithm>
using namespace std;
 
bool cmp(string a, string b) {
    return a + b > b + a;
}
 
string solution(vector<int> numbers) {
    string answer = "";
    vector<string> temp;
    for (int i=0; i<numbers.size(); i++)
        temp.push_back(to_string(numbers[i]));
    sort(temp.begin(), temp.end(), cmp);
    for (int i=0; i<temp.size(); i++){
        if(i==0&&temp[i]=="0"){
            answer+="0";
            break;
        }
         answer+=temp[i];
    }
    return answer;
}

File: programmers/가장 큰 수.java
import java.util.Arrays;
import java.util.Comparator;

class Solution {
    public String solution(int[] numbers) {
        String answer = "";

        String[] arr = new String[numbers.length];
        for (int i = 0; i < numbers.length; i++) {
            arr[i] = (String.valueOf(numbers[i]));
        }


        Arrays.sort(arr, new Comparator<String>() {
            @Override
            public int compare(String s1, String s2) {
                return (s2 + s1).compareTo(s1 + s2);
            }
        });


        if (arr[0].equals("0")) return "0";

        for (int i = 0; i < arr.length; i++) {
            answer += arr[i];
        }

        return answer;
    }
}
